iroh-net/src/magicsock/node_map/node_state.rs

```diff
@@ -128,6 +128,10 @@ pub(super) struct NodeState {
     last_call_me_maybe: Option<Instant>,
     /// The type of connection we have to the node, either direct, relay, mixed, or none.
     conn_type: Watchable<ConnectionType>,
+    /// Whether the conn_type was observed to be `Direct` before.
+    ///
+    /// Used for metric reporting.
+    was_conn_direct_before: bool,
```

Comment:
should we put it behind the metrics flag?

Reply:
The `before` is a bit confusing because it sounds like we are checking _immediately_ before (which we aren't), what about `has_been_direct` or `has_ever_been_direct`?